spec/datetimeformat.html

```diff
@@ -247,37 +251,98 @@
               1. Let _v_ be _v_ modulo 12.
-              1. If _v_ is equal to the value of _tm_.[[<_p_>]], let _pm_ be *false*; else let _pm_ be *true*.
               1. If _v_ is 0 and the value of _dateTimeFormat_.[[hourNo0]] is *true*, let _v_ be 12.
             1. If _f_ is *"numeric"*, then
               1. Let _fv_ be FormatNumber(_nf_, _v_).
             1. Else if _f_ is *"2-digit"*, then
               1. Let _fv_ be FormatNumber(_nf2_, _v_).
               1. If the *length* property of _fv_ is greater than 2, let _fv_ be the substring of _fv_ containing the last two characters.
             1. Else if _f_ is *"narrow"*, *"short"*, or *"long"*, then let _fv_ be a String value representing _f_ in the desired form; the String value depends upon the implementation and the effective locale and calendar of _dateTimeFormat_. If _p_ is *"month"*, then the String value may also depend on whether _dateTimeFormat_ has a [[day]] internal slot. If _p_ is *"timeZoneName"*, then the String value may also depend on the value of the [[inDST]] field of _tm_, and if the implementation does not have a localized representation of _f_, then use _f_ itself.
-            1. Replace the substring of _result_ that consists of *"{"*, _p_, and *"}"*, with _fv_.
-        1. If _dateTimeFormat_.[[hour12]] is *true*, then
-          1. If _pm_ is *true*, then
-            1. Let _fv_ be an implementation and locale dependent String value representing "post meridiem".
-          1. Else,
-            1. Let _fv_ be an implementation and locale dependent String value representing "ante meridiem".
-          1. Replace the substring of _result_ that consists of *"{ampm}"*, with _fv_.
+            1. Add new part record { [[type]]: _p_, [[value]]: _fv_ } as a new element of the list _result_.
+          1. Else if _p_ is equal "ampm", then:
+            1. Let _v_ be the value of _tm_.[[hour]].
+            1. If _v_ is greater than 11, then:
+              1. Let _fv_ be an implementation and locale dependent String value representing "post meridiem";
+            1. Else,
+              1. Let _fv_ be an implementation and locale dependent String value representing "ante meridiem".
+            1. Add new part record { [[type]]: *"dayPeriod"*, [[value]]: _fv_ } as a new element of the list _result_.
+            1. Let _literal_ be the substring of _pattern_ from position _beginIndex_, inclusive, to position _endIndex_, inclusive.
+            1. Add new part record { [[type]]: *"literal"*, [[value]]: _literal_ } as a new element of the list _result_.
+          1. Set _nextIndex_ to _endIndex_ + 1.
+          1. Set _beginIndex_ to Call(*%StringProto_indexOf%*, _pattern_, "}", _nextIndex_)
+        1. If _nextIndex_ is less than _length_, then:
+          1. Let _literal_ be the substring of _pattern_ from position _nextIndex_, inclusive, to position _length_, exclusive.
```

Comment:
great catch! Fixed. It should be exclusive of course. (we don't need `}` as part of the trailing literal)